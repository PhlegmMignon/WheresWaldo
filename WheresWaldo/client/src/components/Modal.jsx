import { useEffect } from "react";

export default function Modal({ modalFound, modal, setModal }) {
  const MODAL_FAIL_STYLE = {
    backgroundColor: "#FF0800",
  };

  const MODAL_SUCCESS_STYLE = {
    backgroundColor: "#32cd32",
  };

  useEffect(() => {
    if (modal) {
      let ele = document.getElementById("modalParent");
      ele.style.display = "block";

      let pos = ele.getBoundingClientRect();
      ele.style.left = pos.left - 30 + "px";
      ele.style.top = pos.top + 50 + "px";

      setTimeout(() => {
        setModal(false);
        ele.style.display = "none";
      }, 3000);
    }
  }, [modal]);

  {
    return modalFound ? (
      <div
        id="modalParent"
        style={MODAL_SUCCESS_STYLE}
        className="fixed text-center p-3 z-10 rounded-lg h-14 w-40"
      >
        <div id="modal" className="text-xl">
          Success!
        </div>
      </div>
    ) : (
      <div
        id="modalParent"
        style={MODAL_FAIL_STYLE}
        className="fixed text-center p-3 z-10 rounded-lg h-14 w-40"
      >
        <div id="modal" className="text-xl">
          Try again
        </div>
      </div>
    );
  }
}
