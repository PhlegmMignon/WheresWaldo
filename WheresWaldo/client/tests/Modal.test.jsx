import { render } from "@testing-library/react";
import Modal from "../src/components/Modal.jsx";

describe("Modal unit tests", () => {
  it("Renders modal when character found is false", async () => {
    const { container } = render(<Modal modalFound={false} />);
    expect(container).toMatchSnapshot();
  });

  it("Renders modal when character found is true", async () => {
    const { container } = render(<Modal modalFound={true} />);

    expect(container).toMatchSnapshot();
  });
});
