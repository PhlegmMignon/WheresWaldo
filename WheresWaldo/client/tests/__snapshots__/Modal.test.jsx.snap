// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal unit tests > Renders modal when character found is false 1`] = `
<div>
  <div
    class="fixed text-center p-3 z-10 rounded-lg h-14 w-40"
    id="modalParent"
    style="background-color: rgb(255, 8, 0);"
  >
    <div
      class="text-xl"
      id="modal"
    >
      Try again
    </div>
  </div>
</div>
`;

exports[`Modal unit tests > Renders modal when character found is true 1`] = `
<div>
  <div
    class="fixed text-center p-3 z-10 rounded-lg h-14 w-40"
    id="modalParent"
    style="background-color: rgb(50, 205, 50);"
  >
    <div
      class="text-xl"
      id="modal"
    >
      Success!
    </div>
  </div>
</div>
`;
